<launch>

    <arg name="model"/>
    <arg name="config_path"/>
    <arg name="uri" default="radio://0/100/2M/0xE7E7E7E7E7"/>
    <arg name="cmd_frequency" default="100"/>
    <!--Initial location-->
    <arg name="init_x" default="0"/>
    <arg name="init_y" default="0"/>
    <!--color for visualization-->
    <arg name="color_r" default="1.0"/>
    <arg name="color_g" default="1.0"/>
    <arg name="color_b" default="0.0"/>
    <arg name="color_a" default="1.0"/>

    <group ns="$(arg model)">

        <!-- Simulator -->
        <node pkg="modquad_simulator" type="modquad_sim.py" name="modquad_sim" output="screen">
            <param name="~robot_id" value="$(arg model)"/>
            <param name="~init_x" value="$(arg init_x)" type="double"/>
            <param name="~init_y" value="$(arg init_y)" type="double"/>
        </node>

        <!--position controller-->
        <include file="$(find crazyflie_controller)/launch/crazyflie2.launch">
            <arg name="frame" value="$(arg model)"/>
        </include>


        <!--mesh publisher for rviz visualization-->
        <node pkg="mesh_visualization"
              type="mesh_visualization"
              name="mesh_visualization"
              output="screen">
            <param name="mesh_resource" value="package://crazyflie_description/meshes/crazyflie2.dae"/>
            <param name="color/r" value="$(arg color_r)"/>
            <param name="color/g" value="$(arg color_g)"/>
            <param name="color/b" value="$(arg color_b)"/>
            <param name="color/a" value="$(arg color_a)"/>
            <remap from="~input" to="odom"/>
        </node>

    </group>
</launch>
